aten/src/ATen/core/Allocator.h

```diff
@@ -133,3 +133,26 @@ struct CAFFE2_API InefficientStdFunctionContext {
 };
 
 } // namespace at
+
+namespace caffe2 {
+
+/** Set the allocator for DeviceType `t` without taking ownership to avoid
```

Comment:
Heh, thank you for updating the comment, but for those folks who didn't review this PR, this comment is very oblique sounding. Just a few words can help, reframing the comment as guidance for where such an allocator might come from. Try:

```
/** Set the allocator for DeviceType `t`. The passed in allocator pointer is
   expected to have static lifetime; this function does NOT take ownership
   of the raw pointer. (The reason for this is to prevent existing pointers
   to an allocator of a particular device from being invalidated when SetAllocator is called.)
```

Reply:
And I guess, while we're on the subject of comments, probably worth mentioning that this function isn't thread safe.